tensorflow/stream_executor/cuda/cuda_blas.cc

```diff
@@ -1692,6 +1668,16 @@ bool CUDABlas::DoBlasGemm(Stream *stream, blas::Transpose transa,
                       "precondition violation";
     }
   }
+
+#if CUBLAS_VER_MAJOR >= 11
+  ScopedCublasMathMode math_mode{blas_};
```

Comment:
For here and other places, you need to hold a lock on `mu_` to access `blas_`. Since `DoBlasInternalImpl` already grabs a lock, probably the best thing to do is to keep the `use_tensor_ops` parameter, or create a new `cublasMath_t` parameer.

Reply:
Done.